### DISCUSSION TITLE
Help get my (working) package to compile with R-universe

### DISCUSSION THREAD

**Discussion by Jean-Romain** [r-universe-org/help#391] [Q&A]
Hi, I just discovered r-universe. Really cool! I'm trying to get my package [lasR](https://github.com/r-lidar/lasR) to compile on r-universe on [r-lidar.r-universe.dev](https://r-lidar.r-universe.dev/builds). The package is not on CRAN but:

- The package links to `gdal` and uses the same `configure` script than the `sf` package
- The package has the same `SystemRequierements` than the `sf` package.
- The package passes all test on all plateform with github action
- The package compiles with CRAN winbuilder and macos builder
- The package works on all plateform for users

Yet ,I have tons of non reproducible error with r-universe (see https://github.com/r-universe/r-lidar/actions/runs/8523541540)  usually different from platform to platform. Each time it seems related to GDAL but the pipeline seems to work for `sf` (see https://github.com/r-universe/r-spatial/actions/runs/8526181519)

I'm kindly asking for some help to get things working.

Regards

**Comment by jeroen** [comment]
To fix the Windows build you need to update Makevars.ucrt from sf too: https://github.com/r-spatial/sf/blob/main/src/Makevars.ucrt
In particular the call to `PKG_LIBS = $(shell pkg-config --libs gdal geos proj)` is important.

**Reply by Jean-Romain** [reply]
This did the job for windows :+1:. Thank you very much for your quick answer. Much appreciated. I'm still having issues on macOS and webassembly.

On webassembly it failed at finding libraries
On macOS release it worked but a units test failed. This goes on my side.
On macOS oldrel it failed at the `configure` stage
```
 checking GDAL: checking whether PROJ is available for running:... no
configure: error: OGRCoordinateTransformation() does not return a coord.trans: PROJ not available?
```

**Reply by Jean-Romain** [reply]
Update on the issue. I badly read the log. Compilation on macOS is working. This is the cross compilation for ARM 64 that is failling

```
./configure: line 3665: ./gdal_proj: Bad CPU type in executable
```

**Reply by Jean-Romain** [reply]
With that in mind I found a fix you wrote for sf https://github.com/r-spatial/sf/pull/2269

Thank  you
